webapp/channels/src/components/common/hooks/use_desktop_notification_permission.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useState} from 'react';
+
+import {type NotificationPermissionNeverGranted} from 'utils/notifications';
+import {isNotificationAPISupported} from 'utils/notifications';
+import {isDesktopApp} from 'utils/user_agent';
+
+type DesktopNotificationPermission = Exclude<NotificationPermission, typeof NotificationPermissionNeverGranted> | undefined;
+
+// This is a hook that is used to request notification permission for desktop app
+// it also returns the permission state. We use this as a workaround for Bug with Electron - https://github.com/electron/electron/issues/11221
+// tl;dr Electron always show 'granted' when queries for Notification.permission, hence this workaround
+export function useDesktopNotificationPermission() {
+    const [desktopResolvePermission, setDesktopResolvePermission] = useState<DesktopNotificationPermission>(undefined);
+
+    useEffect(() => {
+        if (!isDesktopApp()) {
+            return;
+        }
+
+        if (!isNotificationAPISupported()) {
+            return;
+        }
+
+        async function requestNotificationPermission() {
+            // Based on Electron's notification permission it will have following states
+            // - allowed - No further action needed
+            // - denied permanently - No further action needed
+            // - denied (temporary) - In this case, electron notification permission dialog is shown with requestPermission()
+            const permission = await Notification.requestPermission();
+            setDesktopResolvePermission(permission as DesktopNotificationPermission);
+        }
+
+        requestNotificationPermission();
```

Comment:
5/5: If we want to prompt the user again to convince them to enable notifications after they've denied it previously, I don't think we should do it by showing the notification prompt automatically. `Notification.requestPermission` interrupts whatever the user is doing and makes them immediately provide an answer even if they're doing something important, so I think it shouldn't be triggered automatically if we can avoid it.

If we want to prompt the user for permission again, I think it should be done using something less intrusive like the announcement bar where the user doesn't need to interact with it immediately

Reply:
User have the option for permanently disable.